docs/_docs/security/cache-encryption-key-rotation.adoc

```diff
@@ -0,0 +1,147 @@
+
+- link:#command-line-tool[command line tool]
+- link:#jmx[JMX]
+- link:#from-code[from code]
+
+=== Command line tool
+
+Ignite ships a `control.sh|bat` script, located in the `$IGNITE_HOME/bin` folder, that acts as a tool to manage the
+cache encryption key change process from the command line. The following commands can be used with `control.sh|bat`:
+
+[source,shell]
+----
+# View the encryption key identifiers of the cache group.
+control.sh|bat --encryption cache_key_ids cacheGroupName
+
+# Change the cache encryption key.
+control.sh|bat --encryption change_cache_key cacheGroupName
+----
+
+=== JMX
+
+You can also change the cache encryption key via the `EncryptionMXBean` interface:
+
+[cols="1,1",opts="header"]
+|===
+|Method | Description
+|changeCacheGroupKey(String cacheOrGrpName) | Starts cache encryption key change process.
+|===
+
+=== From code
+
+The cache encryption key change process can be managed programmatically:
+
+[tabs]
+--
+tab:Java[]
+
+[source, java]
+----
+include::{javaCodeDir}/TDE.java[tags=cache-group-key-rotation, indent=0]
+----
+--
+
+== Managing the re-encryption process
+
+Re-encrypting existing data can take a long time. This is a fault-tolerant operation that automatically continues after a node restart.
+The previous encryption key is automatically removed when all local partitions are encrypted with the new key, and the last link:persistence/native-persistence#write-ahead-log[Write-Ahead Log] segment, which may contain entries encrypted with the previous key, is removed from disk.
+
+NOTE: Re-encryption uses link:persistence/native-persistence#write-ahead-log[Write-Ahead Log] for physical recovery and may affect performance of cache operations.
+
+There are several options to manage the performance impact of re-encryption:
+
+* Limit the rate of re-encryption by using a configuration parameter or by using the CLI at runtime.
+* Temporarily suspend re-encryption using CLI command.
+
+Ignite 2.10 introduces a new configuration section `EncryptionConfiguration`, that is part of `DatastorageConfiguration`.
+[cols="1,1,1",opts="header"]
+|===
+|Property | Default value | Description
+|reencryptionRateLimit | 0 (unlimited) | Re-encryption rate limit in megabytes per second.
+|reencryptionBatchSize | 100 | The number of pages that is scanned during re-encryption under checkpoint lock.
+|===
+
+=== Using XML configuration to limit re-encryption rate
+[source, xml]
+----
+include::code-snippets/xml/tde.xml[tags=ignite-config;!discovery;!encryption;!cache;!discovery, indent=0]
+----
+
+=== Using CLI to control re-encryption
+
+The `control.sh|bat` script provides the ability to change the re-encryption rate, suspend and resume background re-encryption at runtime.
+
+NOTE: After the node restarts, the suspended background re-encryption will continue automatically, and the rate limit will be set from the local XML configuration (if any) or set to unlimited rate (default).
```

Comment:
'...re-encryption will continue automatically' --> '...re-encryption will be continued automatically'
'...and the rate limit will be set to 'unlimited' (by default), or taken from the local XML configuration (if any).'